velox/dwio/dwrf/test/E2EWriterTest.cpp

```diff
@@ -374,6 +374,93 @@ TEST_F(E2EWriterTest, E2E) {
   dwrf::E2EWriterTestUtil::testWriter(*leafPool_, type, batches, 1, 1, config);
 }
 
+TEST_F(E2EWriterTest, DisableLinearHeuristics) {
+  const size_t batchCount = 100;
+  size_t size = 3000;
+
+  HiveTypeParser parser;
+  auto type = parser.parse(
+      "struct<"
+      "bool_val:boolean,"
+      "byte_val:tinyint,"
+      "short_val:smallint,"
+      "int_val:int,"
+      "long_val:bigint,"
+      "float_val:float,"
+      "double_val:double,"
+      "string_val:string,"
+      "binary_val:binary,"
+      "timestamp_val:timestamp,"
+      "array_val:array<float>,"
+      "map_val:map<int,double>,"
+      "map_val:map<bigint,double>," /* this is column 12 */
+      "map_val:map<bigint,map<string, int>>," /* this is column 13 */
+      "struct_val:struct<a:float,b:double>"
+      ">");
+
+  auto batches = dwrf::E2EWriterTestUtil::generateBatches(
+      type, batchCount, size, /* seed */ 1411367325, *leafPool_);
+
+  auto config = std::make_shared<dwrf::Config>();
+  config->set(dwrf::Config::ROW_INDEX_STRIDE, static_cast<uint32_t>(1000));
+  config->set(dwrf::Config::FLATTEN_MAP, true);
+  config->set(
+      dwrf::Config::MAP_FLAT_COLS,
+      {12, 13}); /* this is the second and third map */
+  config->set<uint64_t>(dwrf::Config::COMPRESSION_BLOCK_SIZE_MIN, 64UL);
+  config->set<uint64_t>(dwrf::Config::STRIPE_SIZE, 25UL * 1024 * 1024);
+
+  // default mode writer
+  dwrf::E2EWriterTestUtil::testWriter(*leafPool_, type, batches, 3, 3, config);
+
+  // disable linear heuristics
+  config->set(dwrf::Config::LINEAR_STRIPE_SIZE_HEURISTICS, false);
```

Comment:
So this test just make sure there is no failure if we turn on/off the hueristics flag? thanks!